ui/pages/Popup.tsx

```diff
@@ -88,6 +90,12 @@ function useConnectPopupMonitor() {
 export function Main(): ReactElement {
   const dispatch = useBackgroundDispatch()
 
+  // Emit an event when the popup page is first loaded.
+  const currentAccount = useBackgroundSelector(selectCurrentAddressNetwork)
+  useEffect(() => {
+    dispatch(userActivityEncountered(currentAccount))
+  })
```

Comment:
this useEffect will run every time Main is rerendered because it has no dependency array, from the comment above I feel like it is not the effect you want to get
```suggestion
  useEffect(() => {
    dispatch(userActivityEncountered(currentAccount))
  }, [])
```